src/libraries/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/DSA.cs

```diff
@@ -419,5 +539,22 @@ private AsnWriter WriteSubjectPublicKeyInfo()
             ImportParameters(key);
             bytesRead = localRead;
         }
+
+        public int GetMaxSignatureSize(DSASignatureFormat signatureFormat)
+        {
+            DSAParameters dsaParameters = ExportParameters(false);
+            int qLength = dsaParameters.Q.Length;
+
+            switch (signatureFormat)
+            {
+                case DSASignatureFormat.IeeeP1363FixedFieldConcatenation:
+                    return qLength * 2;
+                case DSASignatureFormat.Rfc3279DerSequence:
+                    // Add 15 for extra ASN.1 headers
+                    return qLength * 2 + 15;
+                default:
+                    throw new ArgumentOutOfRangeException(nameof(signatureFormat));
```

Comment:
A) I think we want to add a protected virtual property or method to get the size of Q (much cheaper than ExportParameters), which can default to calling ExportParameters.

B) Prefer throwing the argument before calling that member, since ExportParameters will generate a key if it hasn't already happened.